cpp/include/cudf/reduction.hpp

```diff
@@ -122,6 +139,31 @@ std::unique_ptr<column> segmented_reduce(
   null_policy null_handling,
   rmm::mr::device_memory_resource* mr = rmm::mr::get_current_device_resource());
 
+/**
+ * @brief  Compute reduction of each segment in the input column with an initial value. Only SUM,
+ * PRODUCT, MIN, MAX, ANY, and ALL aggregations are supported.
+ *
+ * @param segmented_values Column view of segmented inputs
+ * @param offsets Each segment's offset of @p segmented_values. A list of offsets with size
+ *                `num_segments + 1`. The size of `i`th segment is `offsets[i+1] - offsets[i]`.
+ * @param agg Aggregation operator applied by the reduction
+ * @param output_dtype  The output precision
+ * @param null_handling If `INCLUDE`, the reduction is valid if all elements in a segment are valid,
+ *                      otherwise null. If `EXCLUDE`, the reduction is valid if any element in the
+ *                      segment is valid, otherwise null.
+ * @param init The initial value of the reduction
+ * @param mr Device memory resource used to allocate the returned scalar's device memory
+ * @returns Output column with results of segmented reduction.
+ */
+std::unique_ptr<column> segmented_reduce(
+  column_view const& segmented_values,
+  device_span<size_type const> offsets,
+  segmented_reduce_aggregation const& agg,
+  data_type output_dtype,
+  null_policy null_handling,
+  scalar const& init,
```

Comment:
Similar to above, use `std::optional<std::reference_wrapper<scalar const>> init`.